### ISSUE TITLE
Please provide a version of "eventually" that is not reliant on "experimental" features of Kotlin

### ISSUE THREAD

**Issue by xenomachina** [kotest/kotest#2149]
`eventually` relies on `@ExperimentalTime`. Apparently [this means that even patch releases of kotlinx-coroutines can break binary backwards compatibility](https://github.com/Kotlin/kotlinx.coroutines/issues/2565).  This, in fact, happened with the update from 1.4.2 to 1.4.3 because the mangled name of `delay(Duration)` changed. You can verify this by getting the two jars, kotlinx-coroutines-core-jvm-1.4.3.jar and kotlinx-coroutines-core-jvm-1.4.2.jar, and using javap on DelayKt.class:

``` sh
$ for i in */kotlinx/coroutines/DelayKt.class; do echo "==== $i ===="; javap $i; done
==== kotlinx-coroutines-core-jvm-1.4.2.jar/kotlinx/coroutines/DelayKt.class ====
Compiled from "Delay.kt"
public final class kotlinx.coroutines.DelayKt {
  public static final java.lang.Object awaitCancellation(kotlin.coroutines.Continuation<?>);
  public static final java.lang.Object delay(long, kotlin.coroutines.Continuation<? super kotlin.Unit>);
  public static final java.lang.Object delay-p9JZ4hM(double, kotlin.coroutines.Continuation<? super kotlin.Unit>);
  public static final kotlinx.coroutines.Delay getDelay(kotlin.coroutines.CoroutineContext);
  public static final long toDelayMillis-LRDsOJo(double);
}
==== kotlinx-coroutines-core-jvm-1.4.3.jar/kotlinx/coroutines/DelayKt.class ====
Compiled from "Delay.kt"
public final class kotlinx.coroutines.DelayKt {
  public static final java.lang.Object awaitCancellation(kotlin.coroutines.Continuation<?>);
  public static final java.lang.Object delay(long, kotlin.coroutines.Continuation<? super kotlin.Unit>);
  public static final java.lang.Object delay-VtjQ1oo(double, kotlin.coroutines.Continuation<? super kotlin.Unit>);
  public static final kotlinx.coroutines.Delay getDelay(kotlin.coroutines.CoroutineContext);
  public static final long toDelayMillis-LRDsOJo(double);
}
```
Note `delay-p9JZ4hM` vs `delay-VtjQ1oo`.

So `eventually` from a build of kotest compiled against 1.4.2 will break at runtime if used with 1.4.3.
 
This is particularly bad because `eventually` calls `delay(Duration)` nondeterministically: if the block succeeds on the first attempt, it doesn't call it. This resulted in us having some tests that mysteriously started failing in CI, but only some of the time.

We'd like to use `eventually` (it's quite useful!), but want to avoid using ExperimentalTime, at least until it stabilizes. Would it be possible to make a version that uses a non-ExperimentalTime way to specify the timeout?

**Comment by sksamuel** [comment]
Yes fair point. 4.4.2 is being released now with 1.4.3 so that should keep you good for the time being while we add equivs.

**Comment by sksamuel** [comment]
4.6.0 has been released with a new eventually module.
This eventually is much more configurable and does not rely on experimental durations.
Add module `kotest-framework-concurrency` to your build.
@xenomachina
